railties/test/commands/credentials_test.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+require "isolation/abstract_unit"
+require "env_helpers"
+require "rails/command"
+require "rails/commands/credentials/credentials_command"
+
+class Rails::Command::CredentialsCommandTest < ActiveSupport::TestCase
+  include ActiveSupport::Testing::Isolation, EnvHelpers
+
+  def setup
+    build_app
+  end
+
+  def teardown
+    teardown_app
+  end
+
+  test "edit command does not add master key to gitignore when already exist" do
+    run_edit_command
+
+    Dir.chdir(app_path) do
+      gitignore = File.read(".gitignore")
+      assert_equal 1, gitignore.scan(%r|config/master\.key|).length
+    end
+  end
+
+  private
+    def run_edit_command(editor: "cat")
+      switch_env("EDITOR", editor) do
+        rails "credentials:edit"
+      end
+    end
+end
```

Comment:
There's a separate PR in porting over the `test/commands/secrets_test.rb` to here too, if you're interested 😊

Reply:
Sure! I will do it :smile_cat: